spec/spec_helper.cr

```diff
@@ -15,12 +12,26 @@ def new_test_context(verb = "GET", path = "/api/test")
   {backing_io, HTTP::Server::Context.new(request, response)}
 end
 
-DppmRestApi.run Socket::IPAddress::LOOPBACK, DPPM::Prefix.default_dppm_config.port, __DIR__
+Kemal.config.env = "test"
+
+# Set up the mock permissions.json
 
-# Because the API key for the "normal user" is automatically generated, we
-# need to update the configuration to match the key that was just generated.
-usr = DppmRestApi.permissions_config.users.find { |user| user.name == "Jim Oliver" }.not_nil!
-DppmRestApi.permissions_config.users.delete usr
-usr.api_key_hash = Scrypt::Password.create NORMAL_USER_API_KEY
-DppmRestApi.permissions_config.users << usr
-DppmRestApi.permissions_config.write_to PERMISSION_FILE
+# the location
+macro permissions_file!
+  Path[__DIR__, "permissions.json"]
+end
```

Comment:
The previous `PERMISSION_FILE` constant was better, a macro isn't needed here.

Reply:
Assigning this to a constant broke things elsewhere due to `Path.[]` being a `Call` as opposed to `StringInterpolation` in macro-land. However, that issue seems to have cleared up.